lib/writer.js

```diff
@@ -33,142 +36,170 @@
       else if (tp === "Infinity") return (it.negative ? "-" : "") + "Infinity";
       else if (tp === "NaN") return "NaN";
       else if (tp === "number") return it.value;
-      else if (tp === "sequence") return "[]";
       else return `"${it.value}"`;
     };
+    function default_(def) {
+      let ret = `${def.trivia.assign}=`;
+      if (def.type === "sequence") {
+        ret += `${def.trivia.open}[${def.trivia.close}]`;
+      }
+      else ret += `${def.trivia.value}${const_value(def)}`
+      return ret;
+    }
     function argument(arg) {
       let ret = extended_attributes(arg.extAttrs);
-      if (arg.optional) ret += "optional ";
+      if (arg.optional) ret += `${arg.optional.trivia}optional`;
       ret += type(arg.idlType);
-      if (arg.variadic) ret += "...";
-      ret += ` ${arg.escapedName}`;
-      if (arg.default) ret += ` = ${const_value(arg.default)}`;
+      if (arg.variadic) ret += `${arg.variadic.trivia}...`;
+      ret += `${arg.trivia.name}${arg.escapedName}`;
+      if (arg.default) ret += default_(arg.default);
+      if (arg.separator) ret += `${arg.separator.trivia}${arg.separator.value}`;
       return ret;
     };
+    function identifier(id) {
+      let ret = id.trivia + id.value;
+      if (id.separator) ret += id.separator.trivia + id.separator.value;
+      return ret;
+    }
     function make_ext_at(it) {
       context.unshift(it);
-      let ret = it.name;
+      let ret = it.trivia.name + it.name;
       if (it.rhs) {
-        if (it.rhs.type === "identifier-list") ret += `=(${it.rhs.value.join(",")})`;
-        else ret += `=${it.rhs.value}`;
+        if (it.rhs.type === "identifier-list") ret += `${it.rhs.trivia.assign}=${it.rhs.trivia.open}(${it.rhs.value.map(identifier).join("")}${it.rhs.trivia.close})`;
+        else ret += `${it.rhs.trivia.assign}=${it.rhs.trivia.value}${it.rhs.value}`;
       }
-      if (it.arguments) ret += `(${it.arguments.length ? it.arguments.map(argument).join(",") : ""})`;
+      if (it.signature) ret += `${it.signature.trivia.open}(${it.signature.arguments.map(argument).join("")}${it.signature.trivia.close})`;
+      if (it.separator) ret += `${it.separator.trivia}${it.separator.value}`;
       context.shift(); // XXX need to add more contexts, but not more than needed for ReSpec
       return ret;
     };
     function extended_attributes(eats) {
-      if (!eats || !eats.length) return "";
-      return `[${eats.map(make_ext_at).join(", ")}]`;
+      if (!eats) return "";
+      return `${eats.trivia.open}[${eats.items.map(make_ext_at).join("")}${eats.trivia.close}]`;
     };
 
     const modifiers = "getter setter deleter stringifier static".split(" ");
```

Comment:
I know this is no in the change sent but:
```
 const modifiers = "getter setter deleter stringifier static".split(" ");
```

Is kinda silly. Let's just do:

```JS
 const modifiers = ["getter",  "setter", "deleter", "stringifier", "static"];
```

Also, if they are not used elsewhere, we can just inline the array.